app/src/main/java/com/lightricks/feedexercise/util/result/Result.kt

```diff
@@ -0,0 +1,20 @@
+package com.lightricks.feedexercise.util.result
+
+/**
+ * Result monad with:
+ * */
+sealed class Result<out E : Any, out T : Any> {
+
+    /**
+     * Success state implementation.
+     * */
+    data class Success<out T : Any>(val value: T) : Result<Nothing, T>()
+
+    /**
+     * Failure state implementation.
+     * */
+    data class Failure<out E : Any>(val error: E) : Result<E, Nothing>()
```

Comment:
Generally a failure is represented by an Exception.

Reply:
IMO its better to replace exception with custom class that can describe what exactly went wrong - also having a custom class gives you freedom to store and pass additional data